api/jobs/handlers.py

```diff
@@ -336,12 +378,22 @@ def get_config(self, _id):
         if c is None:
             c = {}
 
+        # Serve config as formatted json file
         self.response.headers['Content-Type'] = 'application/octet-stream'
         self.response.headers['Content-Disposition'] = 'attachment; filename="config.json"'
 
-        # Serve config as formatted json file
-        encoded = json.dumps({"config": c}, sort_keys=True, indent=4, separators=(',', ': ')) + '\n'
-        self.response.app_iter = StringIO.StringIO(encoded)
+        # Detect if config is old- or new-style.
+        # TODO: remove this logic with a DB upgrade, ref database.py's reserved upgrade section.
+
+        if c.get('config') is not None and c.get('inputs') is not None:
+            # New behavior
+            encoded = json.dumps(c, sort_keys=True, indent=4, separators=(',', ': ')) + '\n'
+            self.response.app_iter = StringIO.StringIO(encoded)
```

Comment:
If this is used multiple places, could we add it as a named encoder in web/encoder.py

Reply:
Done